apps/ops/models/celery.py

```diff
@@ -105,6 +106,23 @@ def timedelta(self):
     def is_success(self):
         return self.state == 'SUCCESS'
 
+    def set_creator(self, data):
+        from ops.models import Job
+
+        request = get_current_request()
+        if request and request.user.is_authenticated:
+            self.creator = request.user
+        elif not self.creator:
+            task = data.get('name')
+            args = data.get('args', [])
+
+            if task == 'ops.tasks.run_ops_job' and args:
+                job_id = args[0]
+                job = Job.objects.filter(id=job_id).first()
+                if job:
+                    self.creator = job.creator
+        self.save()
+
     def __str__(self):
         return "{}: {}".format(self.name, self.id)
 
```

Comment:
I did not find any coding errors nor unusual behaviors in this block of code. It seems it's functioning correctly and efficiently with respect to Python best practices and Django conventions.

No obvious irregularities were found, but there may be room for slight improvements:

* Instead of using `gettext_lazy` everywhere inside the class (everywhere else you'd typically use `gettext`) - this function uses only once could reduce redundancy,
* Use `.to_json()`, when needed rather than calling all methods on object directly (`Job.objects.filter(id=id)`)
* Maybe refactor some method names like `set_creator()` or `is_success()` since they seem redundant given what appears to be done here already

The general structure looks solid so no immediate changes required. However further analysis would benefit from more extensive testing.